lib/hyperswitch.js

```diff
@@ -263,15 +263,21 @@ HyperSwitch.prototype._createFilteredHandler = function(handler, filters, specIn
             }
 
             if (filter.method
-                    && filter.method !== req.method
-                    && !(filter.method === 'get' && req.method === 'head')) {
+                && filter.method !== req.method
+                && !(filter.method === 'get' && req.method === 'head')) {
                 return handlerWrapper(hyper, req);
             }
 
-            return filter.filter(hyper, req, handlerWrapper, filter.options, specInfo);
+            return filter.filter(hyper, req, function(hyper, req) {
```

Comment:
Minor: Might be slightly more readable to define the callback as "next" in a separate var, rather than inline.